File: codes/ruby/chapter06/answer_A39.rb
# 入力
n = gets.to_i
a = []
n.times do
  l, r = gets.split.map(&:to_i)
  a << [r, l]
end

a.sort!

current_time = 0
answer = 0
a.each do |r, l|
  if current_time <= l
    current_time = r
    answer += 1
  end
end

# 出力
puts answer

File: codes/go/chapter06/answer_A39.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var N int
	fmt.Scan(&N)

	A := make([][2]int, N)
	for i := 0; i < N; i++ {
		fmt.Scan(&A[i][0], &A[i][1])
	}

	sort.Slice(A, func(i, j int) bool {
		return A[i][1] < A[j][1]
	})

	var CurrentTime int
	var Answer int
	for i := 0; i < N; i++ {
		if CurrentTime <= A[i][0] {
			CurrentTime = A[i][1]
			Answer += 1
		}
	}

	fmt.Println(Answer)
}
